### PR TITLE
Read all drafts for user

### PR DESCRIPTION
**Pull request by rkondratowicz**
### Change description ###

Retrieve all drafts for given user and type.
The old endpoint uses new repo method as well, but still returns data in old format.

**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/functionalTest/java/uk/gov/hmcts/reform/draftstore/data/DraftStoreDAOTest.java [hunk 1]
```diff
@@ -97,22 +99,20 @@ public void shouldRetrieve() throws SQLException {
...
-        assertThat(document).isEqualTo("{ \"test\":\"1234\"}");
+        assertThat(drafts.size()).isEqualTo(1);
+        assertThat(drafts.get(0).document).isEqualTo("{ \"test\":\"1234\"}");
         assertNoOtherUserDataHasBeenAffected();
     }
 
-    @Test(expected = NoDraftFoundException.class)
-    public void shouldThrowOnRetrieveWhenNoDocumentForUser() throws SQLException {
-        underTest.retrieve(USER_ID, "default");
-    }
```
Comment (rkondratowicz) [inline src/functionalTest/java/uk/gov/hmcts/reform/draftstore/data/DraftStoreDAOTest.java]: repo no longer throw exception, moved to controller

Comment (lgonczar) [reply src/functionalTest/java/uk/gov/hmcts/reform/draftstore/data/DraftStoreDAOTest.java]: I'd replace that test with one that checks if the DAO returns an empty list.

Comment (lgonczar) [reply src/functionalTest/java/uk/gov/hmcts/reform/draftstore/data/DraftStoreDAOTest.java]: It's also worth adding a test checking if we can retrieve multiple drafts.

File: src/main/java/uk/gov/hmcts/reform/draftstore/data/DraftStoreDAO.java [hunk 2]
```diff
@@ -62,15 +62,14 @@ public SaveStatus insertOrUpdate(String userId, String type, String newDocument)
...
-    public String retrieve(String userId, String type) {
-        MapSqlParameterSource params =
+    public List<Draft> readAll(String userId, String type) {
+        return jdbcTemplate.query(
+            "SELECT * FROM draft_document WHERE user_id = :userId AND document_type = :type",
             new MapSqlParameterSource()
                 .addValue("userId", userId)
-                .addValue("type", type);
-
-        List<String> documents = jdbcTemplate.query(QUERY, params, (rs, rowNum) -> rs.getString(1));
```
Comment (lgonczar) [inline src/main/java/uk/gov/hmcts/reform/draftstore/data/DraftStoreDAO.java]: QUERY is probably no longer used

Comment (rkondratowicz) [reply src/main/java/uk/gov/hmcts/reform/draftstore/data/DraftStoreDAO.java]: removed

**Pull request merged**